src/coreclr/jit/importer.cpp

```diff
@@ -4311,6 +4311,32 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_get_IsByRefLike:
```

Comment:
This optimization is (functionally) independent on the byreflike types as generic parameters feature. It can be submitted in separate PR to keep this one smaller.

Reply:
See https://github.com/dotnet/runtime/pull/67852.